### ISSUE TITLE
Make get cache locally cache in the app

### ISSUE THREAD

**Issue by frikky** [Shuffle/Shuffle#1108]
When using e.g. "check cache contains" with a loop (e.g. 100), it will go like this:
1. App runs one time, function runs 100 times
2. In function (100 times), it gets cache from the backend
3. Backend loads in the key from database OR if it's cached, from local cache
4. Backend returns it (100 times)

The point is to alleviate step 3 & 4, as to make the App itself in the "get_cache" function store the key locally. 

In short: move cache closer to the source
